# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1321608750 0
#      Fri Nov 18 09:32:30 2011 +0000
# Node ID 2f42a7b0cf940d967180e58548e981429b4de485
# Parent  ecf0c6bca0c97d2221b6d4653a39af907830d671
Fix MinGW compilation problem with tlhelp32.h

* src/sysdep.cc (w32_set_octave_home): Only use TH32CS_SNAPMODULE32 when it
is defined.

diff --git a/src/sysdep.cc b/src/sysdep.cc
--- a/src/sysdep.cc
+++ b/src/sysdep.cc
@@ -105,18 +105,21 @@ BSD_init (void)
 #define WIN32_LEAN_AND_MEAN
 #include <tlhelp32.h>
 
 static void
 w32_set_octave_home (void)
 {
   std::string bin_dir;
 
-  HANDLE h = CreateToolhelp32Snapshot (TH32CS_SNAPMODULE |
-				       TH32CS_SNAPMODULE32, 0);
+  HANDLE h = CreateToolhelp32Snapshot (TH32CS_SNAPMODULE
+#ifdef TH32CS_SNAPMODULE32
+                                       | TH32CS_SNAPMODULE32
+#endif
+                                       , 0);
 
   if (h != INVALID_HANDLE_VALUE)
     {
       MODULEENTRY32 mod_info;
 
       ZeroMemory (&mod_info, sizeof (mod_info));
       mod_info.dwSize = sizeof (mod_info);
 
